lms/views/admin/organization.py

```diff
@@ -96,6 +95,27 @@ def toggle_organization_enabled(self):
         self.request.session.flash("Updated organization", "messages")
         return {"org": org}
 
+    @view_config(
+        route_name="admin.organizations",
+        request_method="POST",
+        renderer="lms:templates/admin/organizations.html.jinja2",
+    )
+    def search(self):
+        if not any((self.request.params.get(param) for param in ["public_id", "name"])):
+            self.request.session.flash(
+                "Need to pass at least one search criteria", "errors"
+            )
+            return HTTPFound(location=self.request.route_url("admin.organizations"))
+
+        if public_id := self.request.params.get("public_id"):
+            orgs = [self.organization_service.get_by_public_id(public_id)]
```

Comment:
Technically you are not "searching" for a public_id, if you find something that's it.

But I prefer the behavior to get the result back in the same page for both name and ID.